src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -0,0 +1,435 @@
+                    let span = epoch.Elapsed
+                    epoch.Restart()
+                    span
+                interface Serilog.Core.ILogEventSink with
+                    member _.Emit logEvent = logEvent |> function
+                        | MetricEvent (Read stats) -> LogSink.Read.Ingest stats
+                        | MetricEvent (ReadL stats) -> LogSink.ReadL.Ingest stats
+                        | MetricEvent (Write stats) -> LogSink.Write.Ingest stats
+                        | MetricEvent (Resync stats) -> LogSink.Resync.Ingest stats
+                        | MetricEvent (Rollup _) -> ()
+                        | _ -> ()
+
+        /// Relies on feeding of metrics from Log through to Stats.LogSink
+        /// Use Stats.LogSink.Restart() to reset the start point (and stats) where relevant
+        let dump (log : ILogger) =
+            let stats =
+              [ "Read", Stats.LogSink.Read
+                "ReadL", Stats.LogSink.ReadL
+                "Write", Stats.LogSink.Write
+                "Resync", Stats.LogSink.Resync ]
+            let logActivity name count lat =
+                log.Information("{name}: {count:n0} requests; Average latency: {lat:n0}ms",
+                    name, count, (if count = 0L then Double.NaN else float lat/float count))
+            let mutable rows, totalCount, totalMs = 0, 0L, 0L
+            for name, stat in stats do
+                if stat.count <> 0L then
+                    totalCount <- totalCount + stat.count
+                    totalMs <- totalMs + stat.ms
+                    logActivity name stat.count stat.ms
+                    rows <- rows + 1
+            // Yes, there's a minor race here between the use of the values and the reset
+            let duration = Stats.LogSink.Restart()
+            if rows > 1 then logActivity "TOTAL" totalCount totalMs
+            let measures : (string * (TimeSpan -> float)) list = [ "s", fun x -> x.TotalSeconds(*; "m", fun x -> x.TotalMinutes; "h", fun x -> x.TotalHours*) ]
+            let logPeriodicRate name count = log.Information("rp{name} {count:n0}", name, count)
+            for uom, f in measures do let d = f duration in if d <> 0. then logPeriodicRate uom (float totalCount/d |> int64)
+
+
+module private Write =
+    let private writeEventsAsync (log : ILogger) (conn : MessageDbClient) (streamName : string) (version : int64) (events : IEventData<EventBody> array)
+        : Async<MdbSyncResult> = async {
+            let! wr = conn.WriteMessages(streamName, events, version, Async.DefaultCancellationToken) |> Async.AwaitTaskCorrect
+            return
+              match wr with
+              | MdbSyncResult.ConflictUnknown ->
+                  log.Information("MdbTrySync WrongExpectedVersionException writing {EventTypes}, expected {ExpectedVersion}",
+                                  [| for x in events -> x.EventType |], version)
+                  wr
+              | _ -> wr }
```

Comment:
return wr here, remove return on 133 and unindent

Reply:
Yessir